src/utils/analytics.ts

```diff
@@ -0,0 +1,61 @@
+import { AnalyticsBrowser } from "@segment/analytics-next";
+import { ExtensionStorage, TempTransactionStorage } from "./storage";
+import { useState, useEffect } from "react";
+
+const analytics = AnalyticsBrowser.load({
+  writeKey: process.env.PLASMO_PUBLIC_SEGMENT_WRITEKEY
+});
+
+export enum EventType {
+  FUNDED = "FUNDED",
+  CONNECTED_APP = "CONNECTED_APP",
+  LOGIN = "LOGIN",
+  ONBOARDED = "ONBOARDED"
+}
+
+export const trackEvent = async (eventName: EventType, properties: any) => {
+  // first we check if we are allowed to collect data
+  const enabled = await ExtensionStorage.get<boolean>("setting_analytics");
+
+  if (!enabled) return;
+
+  const ONE_HOUR_IN_MS = 3600000;
+
+  // TODO:login is tracked only once and compared to an hour period before logged as another Login event
+  if (eventName === EventType.LOGIN) {
+    const storageTime = await TempTransactionStorage.get(EventType.LOGIN);
```

Comment:
Are you sure you wanted to use `TempTransactionStorage` here @nicholaswma? This resets frequently, sometimes after minutes. As far as I can see, you want it persisted for at least an hour? If so, than I'd recommend using either the extension local storage or `window.localStorage`.

Reply:
Hmm, good point. I had the hour as a placeholder to not track every single time the user hits the main page but I think we can probably expand further on the specs; something like if a user opens the extension, closes it, and opens it again in the span of 10 seconds or so, do we consider that 2 logins?